### ISSUE TITLE
fix: AppBar component uses incorrect roles for nav element and  links

### ISSUE THREAD

**Issue by adamint** [microsoft/fluentui-blazor#1864]
# 🐛 Bug Report

As seen in https://github.com/dotnet/aspire/issues/3397, the AppBar component's nav needs the `navigation` role, and the links within should not have any role.

## 💻 Repro or Code Sample

`AppBar.razor` contains the relevant code.

## 🤔 Expected Behavior

`nav` element needs `role="navigation"`, sublinks (`a`) should not have any role.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/navigation_role#examples

## 🔦 Context

This component fails MAS standards and will not be able to be used in Microsoft applications without the requested change.
